projects/packages/videopress/src/class-block-editor-content.php

```diff
@@ -21,36 +21,200 @@ public static function init() {
+			'<div class="jetpack-videopress-player__wrapper">' .
+				'<iframe ' .
+					'title="' . __( 'VideoPress Video Player', 'jetpack-videopress-pkg' ) . '" ' .
+					'aria-label="' . __( 'VideoPress Video Player', 'jetpack-videopress-pkg' ) . '" ' .
+					'src="%s" ' .
+					'width="%s"' .
+					'height="%s" ' .
+					'frameborder="0" ' .
+					'allowfullscreen data-resize-to-parent="true" allow="clipboard-write">' .
+				'</iframe>' .
+			'</div>' .
+		'</figure>';
+
+		$version = Package_Version::PACKAGE_VERSION;
+		wp_enqueue_script( 'videopress-iframe', 'https://videopress.com/videopress-iframe.js', array(), $version, true );
+
+		return sprintf( $block_template, $src, $width, $height );
+	}
+
+	/**
+	 * Generates a VideoPress video block content with the given guid
 	 *
 	 * @param string  $content Post content.
 	 * @param WP_Post $post Post.
 	 * @return string
 	 */
-	public static function video_block_by_guid( $content, $post ) {
+	public static function videopress_video_block_by_guid( $content, $post ) {
 		if ( isset( $_GET['videopress_guid'], $_GET['_wpnonce'] )
 			&& wp_verify_nonce( sanitize_text_field( wp_unslash( $_GET['_wpnonce'] ) ), 'videopress-content-nonce' )
 			&& current_user_can( 'edit_post', $post->ID )
 			&& '' === $content
 		) {
 			$guid = sanitize_text_field( wp_unslash( $_GET['videopress_guid'] ) );
 
+			$base_url     = 'https://videopress.com/v/' . $guid;
+			$query_params = array(
+				'resizeToParent'  => 'true',
+				'cover'           => 'true',
+				'preloadContent'  => 'metadata',
+				'useAverageColor' => 'true',
+			);
+			$url          = esc_url( add_query_arg( $query_params, $base_url ) );
+
 			if ( ! empty( $guid ) ) {
 				// ref /client/lib/url/index.ts
 				$content = '<!-- wp:videopress/video {"guid":"' . $guid . '"} -->
 				<figure class="wp-block-videopress-video wp-block-jetpack-videopress jetpack-videopress-player">
-					<div class="jetpack-videopress-player__wrapper">
-						https://videopress.com/v/' . $guid . '?resizeToParent=true&amp;cover=true&amp;preloadContent=metadata&amp;useAverageColor=true
-					</div>
+					<div class="jetpack-videopress-player__wrapper">' . $url . '</div>
 				</figure>
 				<!-- /wp:videopress/video -->';
 			}
 		}
 
 		return $content;
 	}
+
+	/**
+	 * Override the standard video short tag to also process videopress files as well.
+	 *
+	 * This will parse the given src and, if it is a videopress file, parse as the
+	 * VideoPress shortcode instead.
+	 *
+	 * @param string $html     Empty variable to be replaced with shortcode markup.
+	 * @param array  $attr     Attributes of the video shortcode.
+	 * @param string $content  Video shortcode content.
+	 * @param int    $instance Unique numeric ID of this video shortcode instance.
+	 *
+	 * @return string
+	 */
+	public static function video_shortcode_override( $html, $attr, $content, $instance ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```

Comment:
This is the same code present on the Jetpack VideoPress shortcode class, right? https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/videopress/shortcode.php#L189

Reply:
Yes, it is. Duplicating it to map it to the new shortcode instead of the jetpack one.